Реализована функциональность согласно ТЗ №6

OlegDyuzhin opened the pull request.

OlegDyuzhin committed:
Реализована функциональность согласно ТЗ №6

OlegDyuzhin commented:
Папка tests переименована в test т.к. гитхаб искал папку ./test

avfyodorov commented on src/model/util/Node.java:
```diff
@@ -0,0 +1,33 @@
+package model.util;
+
+public class Node<E> {
```
Этот класс можно сделать внутренним прямо в менеджере истории, потому что больше он нигде не нужен. 
Необязательно, на Ваше усмотрение.

OlegDyuzhin replied:
Перенес в кастомную мапу

avfyodorov commented on src/model/util/Node.java:
```diff
@@ -0,0 +1,33 @@
+package model.util;
+
+public class Node<E> {
```
public class Node\<E\> {
    private final E item;

Можно было бы упростить и реализовать этот класс без обобщений (дженериков). Благодаря наследованию всё и так прекрасно работает.  
```
	public static class Node {
		Task task;
...................
	public Node(Task task, Node prev, Node next) {
```	
Необязательно, на Ваше усмотрение.

OlegDyuzhin replied:
оставил как есть

avfyodorov replied:
Оставляйте, конечно. Вы же учитесь. Просто имейте в виду, что обобщение типа имеет практический смысл, когда в рамках одного алгоритма обрабатываются разные ветки классов, например string и integer. 
А когда классы-прямые наследники, то проще объявить тип родителя, в нашем случае Task.

avfyodorov commented on src/controllers/InMemoryHistoryManager.java:
```diff
@@ -1,25 +1,85 @@
+    }
+
+
+    private static class CustomLinkedHashMap<E extends Task> {
```
```
    private static class CustomLinkedHashMap<E extends Task> {
```
В принципе, можно было и не оформлять отдельным классом, а сделать три-четыре приватных метода. 

Необязательно, на Ваше усмотрение.

OlegDyuzhin replied:
Я не знаю какая будет дальнейшая реализация, но кажется этот класс можно было бы вынести как отдельный, с нодой внутри, тогда его можно будет использовать не только для хранения Task-ов

avfyodorov replied:
Дальше будет сохранение данных в файл и восстановление из... 
А также работа по протоколу http. 
С историей всё, закончили.

avfyodorov commented on src/controllers/InMemoryTaskManager.java:
```diff
@@ -110,19 +110,25 @@ public void updateTask(Task task) {
     public void removeTaskById(int id) {
```
```
public interface TaskManager {
    int getCounterId(TypeTask typeTask);

    ArrayList<Task> getAllTasks();

    ArrayList<Task> getAllEpics();

    ArrayList<Task> getAllSubtasks();
```
Здесь и дальше.
Лучше определить возвращаемый тип как List. Тогда, при реализации данного интерфейса, можно будет применить любой класс, реализующий List. А  не только ArrayList

OlegDyuzhin replied:
переделал

avfyodorov reviewed (commented):
Добрый день, Олег!
Очень хорошая работа, серьёзных замечаний нет, есть уточнения, но и они на Ваше усмотрение.

OlegDyuzhin committed:
Устранены замечания по ревью:
- в интерфейсе TaskManager изменены ArrayList -> List;
- Класс Node перемещен внутрь CustomLinkedHashMap

avfyodorov reviewed (commented):
Добрый день, Олег!

Замечаний нет.
Работа принята.

Оставил комментарии на гитхабе.
